Assets/Countly/Plugins/CountlySDK/Services/UserProfile.cs

```diff
@@ -0,0 +1,682 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Plugins.CountlySDK;
+using Plugins.CountlySDK.Enums;
+using Plugins.CountlySDK.Helpers;
+using Plugins.CountlySDK.Models;
+using Plugins.CountlySDK.Services;
+
+public class UserProfile : AbstractBaseService, IUserProfileModule
+{
+    #region UserProfileData Parameters and Keys
+    internal string Name { get; set; }
+    internal string Username { get; set; }
+    internal string Email { get; set; }
+    internal string Organization { get; set; }
+    internal string Phone { get; set; }
+    internal string PictureUrl { get; set; }
+    internal string Gender { get; set; }
+    internal int BirthYear { get; set; }
+    static string NAME_KEY = "name";
+    static string USERNAME_KEY = "username";
+    static string EMAIL_KEY = "email";
+    static string ORG_KEY = "organization";
+    static string PHONE_KEY = "phone";
+    static string PICTURE_KEY = "picture";
+    static string GENDER_KEY = "gender";
+    static string BYEAR_KEY = "byear";
+    static string CUSTOM_KEY = "custom";
+    readonly string[] NamedFields = { NAME_KEY, USERNAME_KEY, EMAIL_KEY, ORG_KEY, PHONE_KEY, PICTURE_KEY, GENDER_KEY, BYEAR_KEY };
+    #endregion
+
+    internal Dictionary<string, object> CustomDataProperties = new Dictionary<string, object>();
+    internal Dictionary<string, JObject> CustomMods = new Dictionary<string, JObject>();
+    private readonly Countly cly;
+    private readonly CountlyConfiguration config;
+    private readonly CountlyUtils utils;
+    internal readonly RequestCountlyHelper requestHelper;
+    internal readonly EventCountlyService eventService;
+    internal UserProfile(Countly countly, CountlyConfiguration configuration, CountlyLogHelper logHelper, RequestCountlyHelper requestCountlyHelper, CountlyUtils countlyUtils, ConsentCountlyService consentService, EventCountlyService events) : base(configuration, logHelper, consentService)
+    {

```

Comment:
Oh is not the baseservice holding reference to the cly obj. That is interestimg

Reply:
also conf

Reply:
It is ok btw it is just that android and probably ios not like this

Reply:
it doesn't have one for the cly that is correct. it does have one for config but i guess if we would refactor these we would do all of them at the same time for all 'Services' to make sure all is right